### DISCUSSION TITLE
Should entity attack behavior replace standard player behavior?

### DISCUSSION THREAD

**Discussion by Draylar** [Draylar/identity#91] [Q&A] [answered]
Some entities, like Iron Golems, offer custom attack mechanics. Should these replace the standard player attack mechanics (held item used to calculate attack damage) by default?

*For all examples, assume the held stack is not considered for anything like enchantments. Things like Fire Aspect or Sharpness do not apply. The held item damage is not considered at all when attacking, unless the Identity's standard attack function takes it into consideration (for example, Zombies).*

**Example 1: Iron Golem**
The enemy is flung into the air, deals standard Iron Golem damage.

**Example 2: Husk**
The enemy is inflicted with hunger, and then the MobEntity held tool + mob attack damage functionality applies.

**Example 3: Parrot**
The enemy always takes 3 HP of damage.

**Example 4: Cave Spider**
The enemy takes 7 damage in normal mode, 15 damage in hard mode, and is inflicted with poison.

**Comment by pigsteel** [comment]
Yes! Definitely. I vouch for this but on the condition (just an idea) that speed scaling is included. That way it's much more balanced for cases like the iron golem. Alongside this, maybe remove sprinting for some amount of mobs, but keep it for mobs such as piglins and villagers/illagers

**Reply by pigsteel** [reply]
The whole "remove sprinting for some mobs" is because some mobs (as I said, piglins, etc.) have two speeds for different movements, while mobs like zombies or spiders only travel at one speed. Mobs like skeletons and vindicators could trigger their "attacking pose" (raising arms) when sprinting too, since that seems to be a common theme among sprinting mobs.

**Reply by pigsteel** [reply]
Oh! Just did some testing, I guess iron golems can sort of sprint too.

**Comment by notabirb1** [comment]
Yes! That would make a lot more sense. (At least, for hostile mobs, because most passive mobs can't attack)

**Comment by ghost** [comment]
Yes! You should definitely add this as it would make more sense and make some mobs have more of a use as well.

**Comment by James103** [comment]
Maybe allow both player attack and mob attack to go through when you attack? If this is the case, the higher of the two damage amounts will be dealt to the target, and any special effects triggered by the mob attack will go through as well.

**Comment by Ch1216** [comment, marked as answer]
Only replace if the player is barehands, If it has a sword for example return to normal player activity, and for passive mobs. dont do anything (or make it a config)

But if the enemy can hold a sword/item like a husk/zombie do replace while adding the mob entity held tool thing

**Reply by Draylar** [reply]
This is what I was originally leaning towards, and I just implemented it in 1.11.7. Still open to modifying the mechanics in the future if people want Poison-with-swords Spiders or Wither-axes Wither Skeleton.

**Comment by kristibektashi** [comment]
Yes ypu should do that

**Comment by fayeispro** [comment]
Yes. That sounds super cool!
